Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -3156,8 +3156,10 @@ void FrameLoader::updateRequestAndAddExtraFields(ResourceRequest& request, IsMai
     if (shouldUpdate == ShouldUpdateAppInitiatedValue::Yes && localFrame->loader().documentLoader())
         request.setIsAppInitiated(localFrame->loader().documentLoader()->lastNavigationWasAppInitiated());
 
-    if (page && isMainResource)
-        request.setURL(page->chrome().client().applyLinkDecorationFiltering(request.url(), LinkDecorationFilteringTrigger::Navigation));
+    if (page && isMainResource) {
+        bool didFilterLinkDecorationOnNavigation { false };
+        request.setURL(page->chrome().client().applyLinkDecorationFiltering(request.url(), LinkDecorationFilteringTrigger::Navigation, &didFilterLinkDecorationOnNavigation), didFilterLinkDecorationOnNavigation);
```

Comment:
Nit - I think this could be written more clearly, once we make `applyLinkDecorationFiltering` return either a struct or pair:

```
auto [filteredURL, didFilter] = page->chrome().client().applyLinkDecorationFiltering(request.url(), LinkDecorationFilteringTrigger::Navigation);
request.setURL(filteredURL, didFilter);
```